### ISSUE TITLE
preversion not failing

### ISSUE THREAD

**Issue by unional** [npm/npm#15198]
#### I'm opening this issue because:

  - [x] npm is doing something I don't understand.
  - [x] Other (_see below for feature requests_):

#### What's going wrong?
I run validation during `preversion`:

```js
{
  "scripts": {
    "validate": "npm test && npm run lint && npm run dependency-check",
    "preversion": "npm run validate",
    "version": "npm run build",
    ....
  }
}
```

I have a lint issue in the code and when I run `npm run validate`, it is failing correctly. i.e it stops at `npm run lint`

When I run `npm version patch`, the `preversion` script is ran and failed, but it continues to run the `version` script and commit the new version.

I think it should stop the version script and fail instead.

### supporting information:

 - `npm -v` prints: 4.0.2
 - `node -v` prints: 6.9.1

**Comment by legodude17** [comment]
I agree, that would be the consistent behavior.

**Comment by lholmquist** [comment]
I wasn't able to reproduce this issue with this code, https://github.com/lholmquist/validate-test

**Comment by legodude17** [comment]
@unional, can you provide a repro?

**Comment by unional** [comment]
I discovered what's the issue.

The problem is in my script. "preversion" logic is working correctly.

My "verify" script is like this: `npm run coverage && npm run lint && npm run build && npm run dependency-check && git checkout -- dist || true`.

I added the `|| true` in there because I want to swallow the error from `git checkout -- dist` before the first release (I scaffold these) because `git checkout -- dist` will throw an error complaining there is no "dist" in the repo to begin with.

What it turns out the `|| true` is evaluated whenever there is an error and keep moving.

So when `npm run lint` fail, it evals `|| true` and move on to `version` script.

I should be doing `npm run coverage && npm run lint && npm run build && npm run dependency-check && (git checkout -- dist || true)`
